pkg/cmd/addon/util.go

```diff
@@ -40,3 +49,33 @@ func getAddonVersion(addon *extensionsv1alpha1.Addon) string {
 	}
 	return ""
 }
+
+func CheckBeforeDisableAddon(f cmdutil.Factory, addons []string) error {
+	labelSelecotor := util.BuildClusterLabel("", addons)
+	r := f.NewBuilder().
+		Unstructured().
+		AllNamespaces(true).
+		LabelSelector(labelSelecotor).
+		ResourceTypeOrNameArgs(true, []string{util.GVRToString(types.ClusterGVR())}...).
+		ContinueOnError().
+		Latest().
+		Flatten().
+		Do()
+	infos, err := r.Infos()
+	if err != nil {
+		return err
+	} else if len(infos) != 0 {
+		errMsg := "There are addons are being used:\n"
+		for _, info := range infos {
+			var cluster v1alpha1.Cluster
+			if err = runtime.DefaultUnstructuredConverter.FromUnstructured(info.Object.(*unstructured.Unstructured).Object, &cluster); err != nil {
+				return err
+			}
+			errMsg += fmt.Sprintf("name: %s namespace: %s addon: %s\n", printer.BoldRed(info.Name), printer.BoldYellow(info.Namespace), printer.BoldRed(cluster.Labels[constant.ClusterDefLabelKey]))
+		}
+		errMsg += "please delete the cluster(s) first!\n"
```

Comment:
If there are clusters using these addons, notify user these clusters and let user confirm if disable/uninstall.